ddprof-ffi/src/exporter.rs

```diff
@@ -275,8 +258,135 @@ pub unsafe extern "C" fn profile_exporter_send(
         Ok(HttpStatus(response.status().as_u16()))
     }() {
         Ok(code) => SendResult::HttpResponse(code),
-        Err(err) => SendResult::Failure(error_into_buffer(err)),
+        Err(err) => SendResult::Failure(err.into()),
+    }
+}
+
+#[export_name = "ddprof_ffi_SendResult_drop"]
+pub unsafe extern "C" fn send_result_drop(result: SendResult) {
+    std::mem::drop(result)
+}
+
+#[export_name = "ddprof_ffi_Vec_tag_new"]
+pub extern "C" fn vec_tag_new<'a>() -> crate::Vec<Tag<'a>> {
+    crate::Vec::default()
+}
+
+/// Pushes the tag into the vec.
+#[export_name = "ddprof_ffi_Vec_tag_push"]
+pub unsafe extern "C" fn vec_tag_push<'a>(vec: &mut crate::Vec<Tag<'a>>, tag: Tag<'a>) {
+    vec.push(tag)
+}
+
+#[allow(clippy::ptr_arg)]
+#[export_name = "ddprof_ffi_Vec_tag_as_slice"]
+pub extern "C" fn vec_tag_as_slice<'a>(vec: &'a crate::Vec<Tag<'a>>) -> Slice<'a, Tag<'a>> {
+    vec.as_slice()
+}
+
+#[export_name = "ddprof_ffi_Vec_tag_drop"]
+pub extern "C" fn vec_tag_drop(vec: crate::Vec<Tag>) {
+    std::mem::drop(vec)
+}
+
+struct ParseTagsError {
+    message: &'static str,
+}
+
+impl Debug for ParseTagsError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "Tag Error: {}", self.message)
+    }
+}
+
+impl Display for ParseTagsError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "The tags string was malformed.")
+    }
+}
+
+impl Error for ParseTagsError {}
+
+fn parse_tag_chunk(chunk: &str) -> Result<Tag, ParseTagsError> {
+    if let Some(first_colon_position) = chunk.find(':') {
+        if first_colon_position == 0 {
+            return Err(ParseTagsError {
+                message: "Tag cannot start with a colon.",
+            });
```

Comment:
Minor: I think it would be a nice UX if we could include the actual tag that failed to parse in the error message (applies to other error messages added in this PR -- it would be nice to always have at least an excerpt of the actual thing that triggered the failure). 

Otherwise a user needs to go dig what's up, and may get confused on which tags made it where, depending on how complex their setup is, and giving them something they can grep their codebase with here I think may save valuable time.